Sources/_NIOFileSystem/FileSystem.swift

```diff
@@ -861,6 +883,9 @@ extension FileSystem {
             case .success:
                 continue
             case let .failure(errno):
+                if errno == .fileExists {
+                   return self._handleCreateDirectoryFileExists(at: path)
+                }
```

Comment:
This isn't necessary: we're creating child directories here which we'll only do if it didn't already exist.